clients/src/main/java/org/apache/kafka/clients/producer/Callback.java

```diff
@@ -54,5 +56,20 @@ public interface Callback {
      *                      <li>{@link org.apache.kafka.common.errors.UnknownTopicOrPartitionException UnknownTopicOrPartitionException}
      *                  </ul>
      */
-    void onCompletion(RecordMetadata metadata, Exception exception);
+    default void onCompletion(RecordMetadata metadata, Exception exception) {};
```

Comment:
this is a breaking change, need to fix other components:
1. trogdor 
2. storage 
3. connect
4. streams

worth it?
https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-17079/1/cloudbees-pipeline-explorer/